### DISCUSSION TITLE
Apex Test Runner ignores .forceignore (i.e. tries to run non-existent tests)

### DISCUSSION THREAD

**Discussion by brianmfear** [forcedotcom/salesforcedx-vscode#6269] [Ideas]
### Summary

VS Code's Apex Test Runner does not appear to support .forceignore. This makes it impossible to run all tests if you have test classes that you've ignored. I specifically came across this problem while trying to create a repo where some of the work is pulled from a separate repo as a submodule; I cannot remove those .forceignore files from the source path without affecting other work in progress.

### Steps To Reproduce:

1. Create a new project.
1. Add `**/archived/**` to the `.forceignore` file.
1. Create two folders in `force-app/main/default/classes`, `active` and `archived`.
1. Place a unit test class in each of these two folders. Only 1 should deploy, as `archived` is ignored.
1. Clicking the refresh button shows both classes.
1. Clicking the Run All button displays error.

### Expected result

All non-ignored files should have their unit tests run. Alternatively, Apex Test Runner should just use the RunLocalTests command instead of RunSpecifiedTests.

### Actual result

Error in output console: "This class name's value is invalid: XYZ. Provide the name of an Apex class that has test methods."

### Additional information

![image](https://user-images.githubusercontent.com/7635797/149678050-a89b05e4-990f-433b-b7ae-6501000463d8.png)

**Salesforce Extension Version in VS Code**:

Name: Salesforce Extension Pack
Id: salesforce.salesforcedx-vscode
Description: Extensions for developing on the Salesforce Platform
Version: 53.13.0
Publisher: Salesforce
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=salesforce.salesforcedx-vscode

**SFDX CLI Version**:

sfdx-cli/7.120.0 win32-x86 node-v14.17.0

**OS and version**:

Microsoft Windows [Version 10.0.22000.434]

**Comment by mshanemc** [comment]
we resolved this a different way: you can only run tests that are in the org, so only those show up in the test panel
